### ISSUE TITLE
First junior depositor inherits a pre-existing insurance loss (loss-inheritance side of #145; deterministic loss to honest depositors)

### ISSUE THREAD

**Issue by 0x-SquidSol** [dcccrypto/percolator-stake#149]
## Summary

`StakePool::effective_junior_balance()` applies the pool's **current** outstanding insurance loss (`net_loss = total_flushed − total_returned`) to the junior tranche with **no baseline for when the junior cohort began**. So if a pool takes a flush loss while there is no junior tranche (the loss is borne by the global/senior cohort), and a junior **later** deposits, the very next valuation re-assigns that **entire pre-existing loss onto the new junior** — making the incumbent (now-senior) LPs whole at the new junior's expense. The new junior was never present for the loss event, yet deterministically and immediately eats it.

This is the **loss-inheritance** manifestation of the same root cause behind #145 (which is the *recovery-snipe* / gain direction). Both are closed by the same fix; see "Fix options."

## Affected code

- `src/state.rs` — `effective_junior_balance()` computes `net_loss = total_flushed − total_returned` and runs `distribute_loss(junior_balance, gross_senior, net_loss)` with no notion of when the junior cohort started.
- `src/processor.rs` — `process_admin_set_tranche_config` has no guard against enabling tranches while `total_flushed > total_returned`; `process_deposit_junior` has no guard against depositing while a loss is outstanding.

## Mechanism + arithmetic (verified against the production functions)

Mode-0 pool. The loss is incurred *before* any junior exists:

| Step | Action | deposited | flushed | junior_balance | junior_lp | total_lp | total_pool_value |
|---|---|---|---|---|---|---|---|
| 1 | Greg deposits 3,000,000 (plain, non-tranche) | 3,000,000 | 0 | 0 | 0 | 3,000,000 | 3,000,000 |
| 2 | Admin flushes 600,000 (Greg's loss is now real) | 3,000,000 | 600,000 | 0 | 0 | 3,000,000 | **2,400,000** |
| 3 | Admin enables tranches | — | | 0 | 0 | 3,000,000 | 2,400,000 |
| 4 | Jane deposits 1,000,000 as the FIRST junior | 4,000,000 | 600,000 | 1,000,000 | 1,000,000 | 4,000,000 | 3,400,000 |

At step 4, `effective_junior_balance()` at deposit time is 0 (junior_balance == 0), so Jane mints 1:1 (`calc_junior_lp_for_deposit(0, 0, 1,000,000) = 1,000,000`) — it looks like a fair entry. But the moment she has a balance:

- `net_loss = 600,000`; `gross_pool = 4,000,000`; `gross_senior = 3,000,000`
- `distribute_loss(1,000,000, 3,000,000, 600,000)` → junior absorbs first → `junior_loss = 600,000`
- `effective_junior_balance = 1,000,000 − 600,000 = 400,000`
- `senior_balance() = 3,400,000 − 400,000 = 3,000,000`

Withdrawals:
- **Jane (junior, 1,000,000 LP):** `calc_junior_collateral_for_withdraw(1,000,000, 400,000, 1,000,000) = 400,000` → deposited 1,000,000, recovers **400,000 (instant −600,000)**.
- **Greg (senior, 3,000,000 LP):** `calc_senior_collateral_for_withdraw(3,000,000, 3,000,000, 3,000,000) = 3,000,000` → recovers his **full 3,000,000**; his 600,000 loss evaporated.

`Σout = 3,400,000 = vault`, so conservation holds globally and no runtime check fires — but 600,000 was transferred from the honest new junior to the incumbent.

## Impact

The first junior depositor into any pool that has an outstanding (un-returned) insurance loss instantly forfeits up to `min(net_loss, deposit)` of principal, gifted to the incumbent LPs. This is the inverse of the intended "junior is first-loss for *future* events." Unprivileged victim (the depositor); no malice required — an ordinary admin config action (`AdminSetTrancheConfig`) plus a routine deposit triggers it.

## Proof of concept

State/math-level test (repo convention), using the exact production functions. Add as `tests/poc_junior_preexisting_loss.rs`:

```rust
use bytemuck::Zeroable;
use percolator_stake::math::{
    calc_junior_collateral_for_withdraw, calc_junior_lp_for_deposit, calc_senior_collateral_for_withdraw,
};
use percolator_stake::state::StakePool;

fn pool() -> StakePool {
    let mut p = StakePool::zeroed();
    p.is_initialized = 1;
    p.set_discriminator();
    p
}

#[test]
fn first_junior_inherits_preexisting_loss() {
    let mut pool = pool();
    pool.total_deposited = 3_000_000;
    pool.total_lp_supply = 3_000_000;
    pool.total_flushed = 600_000; // Greg's loss is real: total_pool_value == 2,400,000
    assert_eq!(pool.total_pool_value().unwrap(), 2_400_000);

    pool.set_tranche_enabled(true);
    pool.set_junior_fee_mult_bps(20_000);

    let eff_at_deposit = pool.effective_junior_balance();
    assert_eq!(eff_at_deposit, 0);
    let jane_lp = calc_junior_lp_for_deposit(pool.junior_total_lp(), eff_at_deposit, 1_000_000).unwrap();
    assert_eq!(jane_lp, 1_000_000);
    pool.set_junior_balance(1_000_000);
    pool.set_junior_total_lp(jane_lp);
    pool.total_deposited += 1_000_000;
    pool.total_lp_supply += jane_lp;

    let eff_after = pool.effective_junior_balance();
    assert_eq!(eff_after, 400_000); // pre-existing loss reassigned onto the new junior

    let jane_back = calc_junior_collateral_for_withdraw(pool.junior_total_lp(), eff_after, jane_lp).unwrap();
    assert_eq!(jane_back, 400_000); // 1,000,000 in -> 400,000 out, instant -600,000

    let greg_back = calc_senior_collateral_for_withdraw(
        pool.senior_total_lp(), pool.senior_balance().unwrap(), 3_000_000).unwrap();
    assert_eq!(greg_back, 3_000_000); // incumbent made whole at the new junior's expense
}
```

**Reproduction** (verified against the compiled v17 functions): Jane 1,000,000 → 400,000 (−600,000); Greg 3,000,000 → 3,000,000.

## Fix options (this is a should-fix; the remediation is a design choice)

Both close this **and** #145 (deposit-during-outstanding-loss):

1. **Gate tranche deposits while a loss is outstanding** — reject `DepositJunior` (and the senior deposit path) when `total_flushed > total_returned` (and/or block `AdminSetTrancheConfig` enable in that state). Minimal and provably closes both directions. Trade-off: blocks new insurance capital during an open loss window — which for an insurance vault may be exactly when more capital is wanted.
2. **Loss-baseline snapshot** — record the outstanding `net_loss` at junior-cohort start (or per-deposit), so `distribute_loss` only charges the junior for losses incurred *after* they joined. More correct (allows deposits + fair attribution) but more complex (per-cohort loss accounting, and the senior side must be handled symmetrically).

Note: this supersedes #145's "design-call, deferred" classification — the loss-inheritance manifestation is a deterministic loss to honest depositors, not an optional pricing preference, so a fix is warranted.
